src/coreclr/jit/importer.cpp

```diff
@@ -1570,6 +1570,19 @@ GenTree* Compiler::getRuntimeContextTree(CORINFO_RUNTIME_LOOKUP_KIND kind)
     {
         assert(kind == CORINFO_LOOKUP_METHODPARAM || kind == CORINFO_LOOKUP_CLASSPARAM);
 
+        if (compIsForInlining() && (kind == CORINFO_LOOKUP_METHODPARAM))
```

Comment:
Why is the `kind == CORINFO_LOOKUP_METHODPARAM` check needed? Does it mean we won't be able to inline non-generic static methods in generic classes?

Reply:
I was think of this check as a TP check to avoid going into FindWellKnownArg. I don't see why would it prevent from inlining a non-generic static methods in generic classes?

Reply:
Then:
```suggestion
        if (compIsForInlining() && ((kind == CORINFO_LOOKUP_CLASSPARAM) || (kind == CORINFO_LOOKUP_METHODPARAM)))
```

I think the instantiation parameter for a non-generic static method in a generic class is going to have `kind == CORINFO_LOOKUP_CLASSPARAM`, but I could be wrong.

There's still potential quadratic complexity for pathological cases here. Perhaps it would be a bit more elegant to have some `InlineInfo::instArgLclNum` directly? We already recognize it here, so it would just require saving the local:
https://github.com/dotnet/runtime/blob/7c0ed473e76a1861568fa2269f855e2d2e3f16cc/src/coreclr/jit/importer.cpp#L12774-L12787